lib/saluki-components/src/sources/checks/mod.rs

```diff
@@ -35,9 +35,11 @@ use self::check::Check;
 
 mod builder;
 
+mod execution_context;
 #[cfg(feature = "python-checks")]
 use self::builder::python::builder::PythonCheckBuilder;
 use self::builder::CheckBuilder;
+use self::execution_context::ExecutionContext; // FIXME: leaking context for other checks
```

Comment:
Is it actually leaking for other checks? Probably, this execution context is something we would use for other types of checks. Also, it's too early to know 😄